src/components/Filter/Filter.tsx

```diff
@@ -45,6 +45,12 @@ interface IStyles {
   [key: string]: SxProps<Theme>;
 }
 
+function generateRandomKey(length: number) {
+  return Math.random()
+    .toString(36)
+    .substring(2, length + 2);
+}
+
```

Comment:
generateRandomKey(length: number) generates a random string of alphanumeric characters with a specified length, making it useful as a key for encryption or other purposes that require a string of characters, whereas Math. random() only generates a random decimal between 0 and 1. There are not too many pros to changing it. This is my way to solve this. 🤗

Reply:
right, it may be unique but it's not constant. You can just give one of the element's value maybe.

Reply:
using the index as the key is also not always a safe approach, as it can lead to issues if the order of items changes, or if items are added or removed from the list. In these cases, React may end up reusing keys, which can result in unexpected behavior. In our case our menu items

Reply:
I recommended using item's text "kurtarma" or "barinma" for example. Not the index.